megamek/src/megamek/common/TankTrailerHitch.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * MegaMek - Copyright (C) 2002,2003,2004,2005 Ben Mazur (bmazur@sev.org)
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ */
+
+package megamek.common;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+
+/**
+ * Represents a trailer hitch that allows a wheeled or tracked vehicle to tow trailers.
+ *
+ * @see megamek.common.MechFileParser#postLoadInit
+ */
+
+public class TankTrailerHitch implements Transporter {
+
+    // Private attributes, constants and helper functions.
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1193349063084937973L;
+    
+    /**
+     * Is this transporter associated with a front or rear-mounted hitch equipment?
+     */
+    private boolean rearMounted = false;
+    
+    public boolean getRearMounted() {
+        return rearMounted;
+    }
+    
+    /**
+     * The entity being towed by this hitch.
+     */
+    protected int towed = Entity.NONE;
+    transient IGame game;
+
+    /**
+     * The front location that loads trailers externally.
+     */
+    private static final int EXTERIOR_LOCATIONS_FRONT = Tank.LOC_FRONT;
+
+    /**
+     * The rear location that loads trailers externally.
+     */
+    private static final int EXTERIOR_LOCATIONS_REAR = Tank.LOC_REAR;
+
+    /**
+     * The <code>String</code> reported when the hitch is in use.
+     */
+    private static final String NO_VACANCY_STRING = "A trailer is attached";
+
+    /**
+     * The <code>String</code> reported when the hitch is available.
+     */
+    private static final String HAVE_VACANCY_STRING = "One trailer";
+
+    // Protected constructors and methods.
+
+    /**
+     * Get the exterior locations that a trailer covers.
+     * <p/>
+     * Sub-classes are encouraged to override this method.
+     *
+     * @param isRear
+     *            - a <code>boolean</code> value stating if the given location
+     *            is rear facing; if <code>false</code>, the location is front
+     *            facing.
+     * @return an array of <code>int</code> listing the exterior locations.
+     */
+    protected int getExteriorLocs(boolean isRear) {
```

Comment:
This looks like something copied from BAHandles but it's not used anywhere for trailer hitches.

Reply:
Fixed. I think I planned on using it and wound up with something else that worked better for hitches.